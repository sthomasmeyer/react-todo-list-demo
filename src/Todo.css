.Todo {
  margin: 15px;
  padding: 5px;
}

.Todo li {
  color: midnightblue;
  margin-bottom: 5px;
}

.Todo button {
  margin-right: 5px;
  color: ghostwhite;
  background-color: midnightblue;
  font-weight: bolder;
}
