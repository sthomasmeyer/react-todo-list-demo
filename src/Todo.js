import React, { useState } from 'react';
import './Todo.css';

// This React component is designed to render an HTML <div> element.
const Todo = ({ id, task, updateItem, removeItem }) => {
  const [updateTask, setUpdateTask] = useState(task);
  const [isUpdating, setIsUpdating] = useState(false);

  const toggleUpdate = () => {
    setIsUpdating((update) => !update);
  };

  const handleChange = (evt) => {
    setUpdateTask(evt.target.value);
  };

  const handleUpdate = (evt) => {
    evt.preventDefault();
    updateItem(id, updateTask);
    setIsUpdating(false);
  };

  // Remember, the [removeItem()] function expects an 'id' attribute to be passed...
  // to it.
  const remove = () => removeItem(id);

  let jsx = (
    <div className='Todo'>
      <li>{task}</li>
      <button onClick={remove}>X</button>
      <button onClick={toggleUpdate}>Update</button>
    </div>
  );

  if (isUpdating) {
    jsx = (
      <div className='Todo'>
        <form onSubmit={handleUpdate}>
          <input type='text' value={updateTask} onChange={handleChange} />
          <button>Confirm Update</button>
        </form>
      </div>
    );
  }

  return jsx;
};

export default Todo;
