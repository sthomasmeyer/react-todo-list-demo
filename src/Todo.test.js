import React from 'react';
import { render } from '@testing-library/react';
import Todo from './Todo';

// Smoke test:
it('renders without crashing', () => {
  render(<Todo />);
});

it('matches snapshot', () => {
  const { asFragment } = render(<Todo />);
  expect(asFragment()).toMatchSnapshot();
});
