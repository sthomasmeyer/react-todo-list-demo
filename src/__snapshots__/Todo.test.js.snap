// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
  >
    <li />
    <button>
      X
    </button>
    <button>
      Update
    </button>
  </div>
</DocumentFragment>
`;
